src/arena.c

```diff
@@ -1201,18 +1201,18 @@ arena_malloc_hard(tsdn_t *tsdn, arena_t *arena, size_t size, szind_t ind,
 		return NULL;
 	}
 
-	if (likely(size <= SC_SMALL_MAXCLASS)) {
+	if (likely(slab)) {
+		assert(size <= SC_SMALL_MAXCLASS);
 		return arena_malloc_small(tsdn, arena, ind, zero);
 	}
 	return large_malloc(tsdn, arena, sz_index2size(ind), zero);
```

Comment:
I can just put `large_malloc` in the `else` of that `if` then if you feels that's clearer: an `assert(!slab ...)` is tautological, since you would've returned on line 1206 otherwise.

Reply:
Put the large part in the else branch makes sense.